environment.go

```diff
@@ -0,0 +1,150 @@
+package imbue
+
+import (
+	"fmt"
+	"math/bits"
+	"os"
+	"reflect"
+	"strconv"
+
+	"golang.org/x/exp/constraints"
+)
+
+// EnvironmentVariable is a constraint for a type that identifies an environment
+// variable.
+//
+// Environment variables are declared by declaring a type that uses
+// imbue.EnvironmentVariable[T] as its underlying type.
+//
+// The name of the declared type is used as the name of the environment
+// variable.
+//
+// T is the type that is produced by parsing the environment variable's value.
+type EnvironmentVariable[T Parseable] interface {
+	variableOfType(T)
+}
+
+// Parseable is a constraint that identifies the set of types that can be parsed
+// from their string representation, such as in an environment variable.
+type Parseable interface {
+	string |
+		int | int16 | int32 | int64 |
+		uint | uint16 | uint32 | uint64
```

Comment:
Is this odd indentation something go format does?

Reply:
Yeah, it's because I've got that constraint spread across multiple lines... kind of annoying, but as they say:

> [Gofmt's style is no one's favorite, yet gofmt is everyone's favorite.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=8m43s)